code/modules/projectiles/projectile/magic.dm

```diff
@@ -18,7 +18,24 @@
 		if(M.anti_magic_check())
 			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
 			return BULLET_ACT_BLOCK
-		M.death(0)
+		if(isliving(M))
+			var/mob/living/L = M
+			if(L.mob_biotypes & MOB_UNDEAD) //negative energy heals the undead
+				if(L.hellbound && L.stat == DEAD)
+					return BULLET_ACT_BLOCK
+				if(iscarbon(L))
+					var/mob/living/carbon/C = L
+					C.regenerate_limbs()
+					C.regenerate_organs()
+				if(L.revive(full_heal = 1))
```

Comment:
revive with full_heal calls regenerate_limbs and regenerate_organ. full_heal is also a boolean.

Reply:
oh, it does? cool, I'll remove the redundant limb and organ regeneration calls, then